core/chain-configs/src/genesis_config.rs

```diff
@@ -264,6 +266,37 @@ impl GenesisRecords {
     }
 }
 
+pub struct GenesisJsonHasher {
+    digest: sha2::Sha256,
+}
+
+impl GenesisJsonHasher {
+    pub fn new() -> Self {
+        Self { digest: sha2::Sha256::new() }
+    }
+
+    pub fn process_config(&mut self, config: &GenesisConfig) {
+        let mut ser = Serializer::pretty(&mut self.digest);
+        config.serialize(&mut ser).expect("Error serializing the genesis config.");
+    }
+
+    pub fn process_record(&mut self, record: &StateRecord) {
+        let mut ser = Serializer::pretty(&mut self.digest);
+        record.serialize(&mut ser).expect("Error serializing the genesis record.");
+    }
+
+    pub fn process_genesis(&mut self, genesis: &Genesis) {
+        self.process_config(&genesis.config);
+        for record in genesis.records.as_ref() {
+            self.process_record(record)
+        }
+    }
```

Comment:
I'd write this as 

```rust
impl GenesisJsonHasher {
    pub fn process_genesis(genesis: &Genesis) -> CryptoHash {
      let mut hasher = Self::new();
      hasher.process_config(&genesis.config);
      genessis.records.iter().for_each(|it| hasher.process_record(it));
      hasher.finalize()
    }
}
```

`&mut self` here doesn't really make sense, as the only correct way to use this function is starting with an empty hasher and calling `.finalize` in the end. So we might as well just make it impossible to call this the wrong way.